Add fourth part to wxVersionInfo

rcane opened the pull request:
This implements the requested feature from #22452.

In addition to adding the fourth part to wxVersionInfo I also extended the Edge and IE webview backends to include the the new part into their versions.

rcane committed:
Added a 'tweak' component to wxVersionInfo.

Some versions consist of four parts with the last part being called the
tweak or build number. wxVersionInfo is now able to save such a fourth
number.

rcane committed:
Get complete versions of Edge and IE backends.

Since wxVersionInfo supports four parts now, we can return the complete
version from the Edge and IE backends to clients.

vadz commented on interface/wx/versioninfo.h:
```diff
@@ -74,6 +76,13 @@ class wxVersionInfo
+        Get the tweak version, or build number.
+
+        @return Tweak version, or build number.
+    */
```
```suggestion
        @since 3.1.7
    */
```

vadz commented on interface/wx/versioninfo.h:
```diff
@@ -35,6 +35,7 @@ class wxVersionInfo
         @param major The major version component.
         @param minor The minor version component.
         @param micro The micro version component, 0 by default.
+        @param tweak The tweak version component, 0 by default.
```
```suggestion
        @param tweak The tweak version component, 0 by default (this parameter is available since wxWidgets 3.1.7 only).
```

vadz reviewed (commented):
Looks good, thanks, just a couple of doc tweaks (sorry...).

BTW, I've tried to search for "tweak version" (and various variants thereof) but couldn't find any mentions of it. I've never seen this word used in this context myself and [Wikipedia](https://en.wikipedia.org/wiki/Software_versioning) doesn't mention it neither, so I wonder just how well-known it is. IMHO "build" is a much more common term, so maybe we should use it instead?

What do the others think, is anybody here familiar with the term "tweak" in the version context?

vadz commented on include/wx/versioninfo.h:
```diff
@@ -23,15 +23,26 @@ class wxVersionInfo
+    wxVersionInfo(const wxString& name, int major, int minor, int micro,
+                  const wxString& description,
+                  const wxString& copyright = wxString())
+        : wxVersionInfo(name, major, minor, micro, 0, description, copyright)
```
Oh, damn, sorry, I missed this during the review, but we still support C++98 and so can't use delegating constructors, so you'll have to duplicate the above lines here or add `Init()` function, as done in other places.

rcane commented:
As for the word `tweak`, I got that from [cmake](https://cmake.org/cmake/help/latest/variable/CMAKE_VERSION.html). 
`wxVersionInfo` calls the third part `micro` while cmake calls it `patch`. So there is obviously no consensus about that either. But I have no problem with renaming `tweak` to `build`. Or we could call it `nano` so that everybody is confused. :-)

vadz commented:
If you allow me to be not very nice just for a second, CMake is not exactly where I'd look for getting clear and unambiguous terminology from.

FWIW [Microsoft uses](https://docs.microsoft.com/en-us/dotnet/api/system.version) `major.minor[.build[.revision]]` and considering that we currently only use this with Microsoft products, it would, IMO, make sense to use the names compatible with their schema, i.e. either "build" or "revision".

rcane commented:
I was not suggesting that CMake should set any kind of standard here. We just use it as our build system and therefor I was familiar with `tweak` as the name for the fourth part of the version.
But lets just use the Microsoft scheme and call it `revision`.

rcane committed:
Merge branch 'wxWidgets:master' into feature/wxversioninfo

rcane committed:
Removed a delegating constructor.

This is a C++11 feature which we cannot use yet.
Instead we use a Init() method that is called from both constructors.

rcane committed:
Renamed 'tweak' to 'revision'.

'tweak' is not a very well know name for the fourth part of a version.
It seems to be used only by cmake. Calling the fourth part 'revision'
is more common.

vadz commented:
Thanks, merged now with a couple of cosmetic tweaks.
